### DISCUSSION TITLE
[Bug?]: File change breaks Prod if no hard refresh

### DISCUSSION THREAD

**Discussion by MariuzM** [solidjs/solid-start#1151] [Ideas]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

I have Page1 and Page2 if i change something on Page2 and then using Page1 to navigate to Page2 i get error 
```
error loading dynamically imported module: https://****/assets/about-52919eb7.js
```

I guess its due to the hash of the file name change but my current client session still working with V1 until i do hard refresh. 

Is there any plan to work around this?

### Expected behavior 🤔

Page2 should load without issue even if changes have been made and pushed to prod

**Comment by MariuzM** [comment]
Ive tested with Next.js it looks like its doing hard page refresh before continue fetching new components

**Comment by ryansolid** [comment]
Hmm.. a client side navigation has no way of knowing it is out of date, except from the response.. and we fetch JS in parallel. Unless they have setup some background pinging of versioning I don't see how Next could ever know this. My understanding was that Vercel keeps previous deploys around. Its also possible I suppose that that fetch data before JS and in that case they could know, of course that would be very inefficient.

In any case this is an idea and if anyone has ideas on how to implement it we can consider making a request.
